rocketchat-core/src/main/java/com/rocketchat/core/RestImpl.java

```diff
@@ -169,40 +178,42 @@ public void onResponse(Call call, Response response) throws IOException {
         });
     }
 
+    // TODO
+    void getRoomMembers() {
+
+    }
+
+    // TODO
+    void getRoomFavoriteMessages() {
+
+    }
+
+    // TODO
+    void getRoomPinnedMessages() {
+
+    }
+
     void getRoomFiles(String roomId,
                       BaseRoom.RoomType roomType,
                       String offset,
-                      String sortBy,
+                      Attachment.SortBy sortBy,
                       Sort sort,
                       final RoomCallback.GetFilesCallback callback) {
-        checkNotNull(offset,"roomId == null");
-        checkNotNull(offset,"roomType == null");
+        checkNotNull(roomId,"roomId == null");
+        checkNotNull(roomType,"roomType == null");
         checkNotNull(offset,"offset == null");
         checkNotNull(sortBy,"sortBy == null");
-        checkNotNull(sort,"sortDirection == null");
+        checkNotNull(sort,"sort == null");
         checkNotNull(callback,"callback == null");
 
-        String path;
-        switch (roomType) {
-            case PUBLIC:
-                path = "channels.files";
-                break;
-            case PRIVATE:
-                path = "groups.files";
-                break;
-            default:
-                path = "dm.files";
-                break;
-        }
-
-        RequestBody body = new FormBody.Builder()
-                .add("roomId", roomId)
-                .add("offset", offset)
-                .add("sort", "{\"" + sortBy + "\":" + sort.getDirection() + "}")
+        HttpUrl httpUrl = requestUrl(baseUrl, getRestApiMethodNameByRoomType(roomType) + "files")
```

Comment:
Maybe passing `"files"` as a parameter to  `getRestApiMethodNameByRoomType`, I think it will be cleaner

Reply:
Much cleaner! How we should named the second parameter? Is "method" good?

```
private String getRestApiMethodNameByRoomType(BaseRoom.RoomType roomType, String method) {
[...]
}
```

Reply:
I think that "method" is the full name (e.g. channels.files)... What do you think?

Reply:
I think method is ok...